src/functionaltests/java/com/ericsson/ei/utils/FunctionalTestBase.java

```diff
@@ -59,4 +78,28 @@ public void beforeTestClass(TestContext testContext) throws Exception {
     public void afterTestClass(TestContext testContext) throws Exception {
         int debug = 1;
     }
+
+    public boolean waitForEventsToBeProcessed(int eventsCount) {
+        int maxTime = 30;
+        int counterTime = 0;
+        long processedEvents = 0;
+        while (processedEvents < eventsCount && counterTime < maxTime) {
+            processedEvents = countProcessedEvents(database, collection);
+            LOGGER.debug("Have gotten: " + processedEvents + " out of: " + eventsCount);
+            try {
+                TimeUnit.MILLISECONDS.sleep(3000);
+                counterTime += 3;
+            } catch (InterruptedException e) {
+                LOGGER.error(e.getMessage(), e);
+            }
+        }
```

Comment:
A list processed events should be store in some Json List format and compared with an expected events Json list in test case and reported in the failed test case report.

Reply:
Need to discuss weather or not this should be implemented as it is not part of the test case, this function was added just to make sure the test does not continue before all the events have been processed.